### PR TITLE
CC-1878: Allow for case differences while finding missing columns

### PR DESCRIPTION
**Pull request by wicknicks**
Signed-off-by: Arjun Satish <arjun@confluent.io>

### CODE REVIEW

**Comment by wicknicks** [comment]
[clabot:check]

File: src/main/java/io/confluent/connect/jdbc/sink/DbStructure.java [hunk 1]
```diff
@@ -172,6 +172,33 @@ boolean amendIfNecessary(
...
+
+    // check if the missing fields can be located by ignoring case
+    Set<String> columnNamesLowerCase = new HashSet<>();
+    for (String columnName: dbColumnNames) {
+      columnNamesLowerCase.add(columnName.toLowerCase());
+    }
+
+    if (columnNamesLowerCase.size() != dbColumnNames.size()) {
+      log.warn("Table has column names which differ only with case sensitivity. Original "
+          + "columns={}", dbColumnNames);
```
Comment (rhauch) [inline src/main/java/io/confluent/connect/jdbc/sink/DbStructure.java]: Perhaps different wording:

    Table has column names that differ from field names only by case. Original ...

File: src/main/java/io/confluent/connect/jdbc/sink/DbStructure.java [hunk 2]
```diff
@@ -172,6 +172,33 @@ boolean amendIfNecessary(
         missingFields.add(field);
       }
     }
-    return missingFields;
+
```
Comment (rhauch) [inline src/main/java/io/confluent/connect/jdbc/sink/DbStructure.java]: Maybe return here if `missingFields.isEmpty()`?

Comment (wicknicks) [reply src/main/java/io/confluent/connect/jdbc/sink/DbStructure.java]: This list is used to create ALTER table queries. So we need to return the set here.

Comment (rhauch) [reply src/main/java/io/confluent/connect/jdbc/sink/DbStructure.java]: Right. If there are no missing fields at this point, why go through the work of checking case-insensitively? Why not return an empty list here?

**Pull request merged**